### PR TITLE
Enhance generic hash packet sending function

### PR DESCRIPTION
**Pull request by echuawu**
### Description of PR
Enhance generic hash script packet sending function    
    1. In case of a weak nic, when packet is not received, resend the packet
    2. Loop more times to make hash result more stable

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
Some weak nic would failed to send packet incidentally
#### How did you do it?
Send more packets
#### How did you verify/test it?
Run it in internal regression

### CODE REVIEW

**Review by wsycqyz — APPROVED** [review]
lğtm

**Comment by wsycqyz** [comment]
Do other branch need this PR?
Also please re-trigger the PR test.

**Comment by echuawu** [comment]
> Do other branch need this PR? Also please re-trigger the PR test.

Re-triggered. The test had been supported since 202311.

File: ansible/roles/test/files/ptftests/py3/generic_hash_test.py [hunk 1]
```diff
@@ -379,8 +379,17 @@ def check_ip_route(self, pkt, masked_expected_pkt, sending_port):
         send the packet and check it is received by one of the expected ports
         """
         testutils.send_packet(self, sending_port, pkt)
-        port_index, received = testutils.verify_packet_any_port(
-            self, masked_expected_pkt, self.expected_port_list, timeout=0.1)
+        try:
+            port_index, received = testutils.verify_packet_any_port(
+                self, masked_expected_pkt, self.expected_port_list, timeout=0.1)
+        except AssertionError:
```
Comment (bingwang-ms) [inline ansible/roles/test/files/ptftests/py3/generic_hash_test.py:385]: Why not add the for loop at line 380 to make the logic easier? And will the retry cover some real issue?

Comment (echuawu) [reply ansible/roles/test/files/ptftests/py3/generic_hash_test.py:385]: > Why not add the for loop at line 380 to make the logic easier? And will the retry cover some real issue?

It just a exception handling method when the PTF failed to send packet at our shared test server. No need to loop send packet. And as this is dedicate for packet hashing validation, packet retry would not affect the hash result. And we have many other cases for packet forwarding validation as well.

**Review by bingwang-ms — APPROVED** [review]

**Pull request merged**